src/OrchardCore.Modules/OrchardCore.AuditTrail/Handlers/UserEventHandler.cs

```diff
@@ -0,0 +1,136 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.DependencyInjection;
+using OrchardCore.AuditTrail.Extensions;
+using OrchardCore.AuditTrail.Providers;
+using OrchardCore.AuditTrail.Services;
+using OrchardCore.AuditTrail.Services.Models;
+using OrchardCore.Users;
+using OrchardCore.Users.Events;
+using OrchardCore.Users.Handlers;
+
+namespace OrchardCore.AuditTrail.Handlers
+{
+    public class UserEventHandler : ILoginFormEvent, IPasswordRecoveryFormEvents, IRegistrationFormEvents, IUserEventHandler
```

Comment:
As above.

This needs to go into the `Users` module

Reply:
Some interfaces will then also require to be put in an abstractions project. Maybe the descriptors should also be moved to their corresponding modules, in an AuditTrail folder.